doc/sphinx/proof-engine/ltac.rst

```diff
@@ -29,151 +30,208 @@ Syntax
+
+.. todo For the compound tactics, review all the descriptions of evaluation vs application,
+   backtracking, etc. to get the language consistent and simple (refactoring so the common
+   elements are described in one place)
+
+Types of values
+~~~~~~~~~~~~~~~
+
+An |Ltac| value can be a tactic, integer, string, unit (written as "`()`" ) or syntactic value.
+Syntactic values correspond to certain nonterminal symbols in the grammar,
+each of which is a distinct type of value.
+Most commonly, the value of an |Ltac| expression is a tactic that can be executed.
+
+While there are a number of constructs that let you combine multiple tactics into
+compound tactics, there are no operations for combining most other types of values.
+For example, there's no function to add two integers.  Syntactic values are entered
+with the :token:`syn_value` construct.  Values of all types can be assigned to toplevel
+symbols with the :cmd:`Ltac` command or to local symbols with the :tacn:`let` tactic.
+|Ltac| :tacn:`functions<fun>` can return values of any type.
+
+.. todo: there are 36 subsections under "Semantics", which seems like far too many
+
+Syntactic values
+~~~~~~~~~~~~~~~~
+
+.. insertprodn syn_value syn_value
+
+.. prodn::
+   syn_value ::= @ident : ( @nonterminal )
+
+Provides a way to use the syntax and semantics of a grammar nonterminal as a
+value in an :token:`ltac_expr`.  The table below describes the most useful of
+these.  You can see the others by running ":cmd:`Print Grammar` `tactic`" and
+examining the part at the end under "Entry tactic:tactic_arg".
+
+   :token:`ident`
+      name of a grammar nonterminal listed in the table
+
+   :production:`nonterminal`
+      represents syntax described by :token:`nonterminal`.
+
+   .. list-table::
+      :header-rows: 1
+
+      * -  Specified :token:`ident`
+        - Parsed as
+        - Interpreted as
+        - as in tactic
+
+      * - ``ident``
+        - :token:`ident`
+        - a user-specified name
+        - :tacn:`intro`
+
+      * - ``string``
+        - :token:`string`
+        - a string
+        -
+
+      * - ``integer``
+        - :token:`int`
+        - an integer
+        -
+
+      * - ``reference``
+        - :token:`qualid`
+        - a qualified identifier
+        -
+
+      * - ``uconstr``
+        - :token:`term`
+        - an untyped term
+        - :tacn:`refine`
+
+      * - ``constr``
+        - :token:`term`
+        - a term
+        - :tacn:`exact`
+
+      * - ``ltac``
+        - :token:`ltac_expr`
+        - a tactic
+        -
+
+:n:`ltac:(@ltac_expr)` can be used to indicate that the parenthesized item
+should be interpreted as a tactic and not as a term.  The constructs can also
+be used to pass parameters to tactics written in OCaml.  (While all
+of the :token:`syn_value`\s can appear at the beginning of an :token:`ltac_expr`,
+the others are not useful because they will not evaluate to tactics.)
+
+:n:`uconstr : ( @term )` can be used to build untyped terms.
```

Comment:
```suggestion
:n:`uconstr:(@term)` can be used to build untyped terms.
```
Spaces not necessary, and a bit jarring, especially as the `ltac:(@ltac_expr)` above has no spaces.  Currently it looks like:
![image](https://user-images.githubusercontent.com/396076/84083840-60b47980-a9b0-11ea-960b-4b7366c39430.png)
![image](https://user-images.githubusercontent.com/396076/84083848-65792d80-a9b0-11ea-8653-b3b65eb81e1f.png)
![image](https://user-images.githubusercontent.com/396076/84083862-6a3de180-a9b0-11ea-81e6-e49df0dd5873.png)
![image](https://user-images.githubusercontent.com/396076/84083866-6e69ff00-a9b0-11ea-95ec-66246d14fef7.png)
(I think the first and last are fine, and the middle two would be better without spaces)

Reply:
Got it.  BTW, the last one would require a code change to the doc_grammar utility.